internal/fs/type_cache_test.go

```diff
@@ -0,0 +1,421 @@
+			mountConfig.TtlInSeconds)
+		serverCfg.InodeAttributeCacheTTL = serverCfg.DirTypeCacheTTL
+		// We can add more logic here to fill other fileds in serverCfg
+		// from mountConfig here as needed.
+	}(t.serverCfg.MountConfig, &t.serverCfg)
+
+	// Call through.
+	t.fsTest.SetUpTestSuite()
+}
+
+////////////////////////////////////////////////////////////////////////
+// Specific test classes
+////////////////////////////////////////////////////////////////////////
+
+type TypeCacheTestWithMaxSize1MB struct {
+	typeCacheTestCommon
+}
+
+func (t *TypeCacheTestWithMaxSize1MB) SetUpTestSuite() {
+	ttlInSeconds = 30
+	typeCacheMaxSizeMb = 1
+
+	t.typeCacheTestCommon.SetUpTestSuite()
+}
+
+type TypeCacheTestWithZeroCapacity struct {
+	typeCacheTestCommon
+}
+
+func (t *TypeCacheTestWithZeroCapacity) SetUpTestSuite() {
+	ttlInSeconds = 30
+	typeCacheMaxSizeMb = 0
+
+	t.typeCacheTestCommon.SetUpTestSuite()
+}
+
+type TypeCacheTestWithZeroTTL struct {
+	typeCacheTestCommon
+}
+
+func (t *TypeCacheTestWithZeroTTL) SetUpTestSuite() {
+	ttlInSeconds = 0
+	typeCacheMaxSizeMb = 1
+
+	t.typeCacheTestCommon.SetUpTestSuite()
+}
+
+type TypeCacheTestWithInfiniteTTL struct {
+	typeCacheTestCommon
+}
+
+func (t *TypeCacheTestWithInfiniteTTL) SetUpTestSuite() {
+	ttlInSeconds = -1
+	typeCacheMaxSizeMb = 1
+
+	t.typeCacheTestCommon.SetUpTestSuite()
+}
+
+func init() {
+	RegisterTestSuite(&TypeCacheTestWithMaxSize1MB{})
+	RegisterTestSuite(&TypeCacheTestWithZeroCapacity{})
+	RegisterTestSuite(&TypeCacheTestWithZeroTTL{})
+	RegisterTestSuite(&TypeCacheTestWithInfiniteTTL{})
+}
+
+// //////////////////////////////////////////////////////////////////////
+// helpers
+// //////////////////////////////////////////////////////////////////////
+func (t *typeCacheTestCommon) testNoInsertion() {
+	const name1 = "foo"
+	const contents = "taco"
+	var fi fs.FileInfo
+	var err error
+
+	// Create a file object in GCS.
+	fileObject, err := storageutil.CreateObject(
+		ctx,
+		bucket,
+		name1,
+		[]byte(contents))
+
+	ExpectEq(nil, err)
+	ExpectNe(nil, fileObject)
+
+	// Stat-call with file object. It should
+	// pass stat call, bypassing type-cache, as a file.
+	fi, err = os.Stat(path.Join(mntDir, name1))
+
+	ExpectEq(nil, err)
+	AssertNe(nil, fi)
+	ExpectFalse(fi.IsDir())
+
+	// Create a directory object in GCS with same name as the file object.
+	dirObject, err := storageutil.CreateObject(
+		ctx,
+		bucket,
+		name1+"/",
+		[]byte(contents))
+
+	ExpectEq(nil, err)
+	ExpectNe(nil, dirObject)
+
+	// Stat-call with directory object. It should
+	// pass stat call, bypassing type-cache, as a directory.
+	// It works because no entries are inserted in type-cache
+	// in this case.
+	fi, err = os.Stat(path.Join(mntDir, name1) + "/")
+
+	ExpectEq(nil, err)
+	AssertNe(nil, fi)
+	ExpectTrue(fi.IsDir())
+}
+
+// //////////////////////////////////////////////////////////////////////
+// Tests for TypeCacheTestWithMaxEntries1
```

Comment:
nit: Tests for TypeCacheTestWithMaxSize1MB ?

Reply:
done.